lib/ansible/modules/cloud/smartos/nictagadm.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2018, Bruce Smith <Bruce.Smith.IT@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: nictagadm
+short_description: Manage NIC tags on SmartOS systems.
+description:
+  - Create of delete NIC tags on SmartOS systems.
+version_added: "2.5"
+author: Bruce Smith (@SmithX10)
+options:
+    name:
+        description:
+            - NIC tag name.
+        required: true
+    mac:
+        description:
+            - MAC Address to attach the NIC tag to when not using an etherstub. mac and etherstub are mutually exclusive.
+        required: false
+        default: None
+    etherstub:
+        description:
+            - Specifies that the NIC tag will be attached to a created etherstub. etherstub is mutually exclusive with both mtu, and mac.
+        required: false
+        default: false
+    mtu:
+        description:
+            - MTU size of the NIC tag. mtu and etherstub are mutually exclusive.
+        required: false
+        default: None
+    force:
+        description:
+            - When State.Absent is set this switch will use the -f parameter and delete the NIC tag regardless of existing VMs
+        default: false
+    state:
+        description:
+            - Create or delete a SmartOS NIC tag.
+        required: false
+        default: "present"
+        choices: [ "present", "absent" ]
+'''
+
+EXAMPLES = '''
+- name: Create 'strage0' on '00:1b:21:a3:f5:4d'
+  nictagadm: name=storage0 mac=00:1b:21:a3:f5:4d mtu=9000 state=present
+
+- name: Remove 'storage0' NIC tag
+  nictagadm: name=storage0 state=absent force=false
+'''
+
+RETURN = '''
+name:
+    description: NIC tag name
+    returned: always
+    type: string
+    sample: storage0
+mac:
+    description: MAC Address that the nic tag was attached to.
+    returned: always
+    type: string
+    sample: 00:1b:21:a3:f5:4d
+etherstub:
+    description: specifies if the NIC tag will create and attach to an etherstub.
+    returned: always
+    type: boolean
+    sample: False
+mtu:
+    description: specifies which MTU size was passed during the nictagadm add command. mtu and etherstub are mutually exclusive.
+    returned: always
+    type: int
+    sample: 1500
+force:
+    description: Shows if -f was used during the deletion of a NIC tag
+    returned: always
+    type: boolean
+    sample: False
+state:
+    description: state of the target
+    returned: always
+    type: string
+    sample: present
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+import re
+
+
+class NICTAG(object):
```

Comment:
This is a personal preference, but I'd rather not have a class name in all caps.